Added Proxmox Docker ZFS Tutorial

Kenzim opened the pull request:
I have read and understood the Contributor's Certificate of Origin available at the end of 
https://raw.githubusercontent.com/hetzneronline/community-content/master/tutorial-template.md
and I hereby certify that I meet the contribution criteria described in it.
Signed-off-by: Kenzi Marcel - kenzi@kenzim.co.uk

Kenzim commented:
hi @n-se , would you prefer it if the tutorial solely focused on installing using the rescue system instead of kvm console? having both methods is making the tutorial quite long

n-se commented:
Both ways have their use case. Other server vendors may not offer KVM. Since you have already described both ways, we can keep both. If you absolutely want to keep only one of the two because of the length, I would prefer the rescue system method.

By the way, you can also ask the Hetzner staff to copy your ISO to an usb stick and attach it to your server.

Kenzim commented:
I think I will keep both methods then. On a side note - I don't know any other dedicated provider who has a rescue system like this. They usually only have KVM. 
What do you think of the tutorial anyway? anything you think should be changed?

n-se commented:
I will review the tutorial in detail in a few days ;)

n-se commented on tutorials/proxmox-docker-zfs/01.en.md:
```diff
@@ -0,0 +1,405 @@
+
+<img src="images/support-kvm2.jpg" alt="&quot;Ordering a KVM console&quot;" title="Ordering a KVM console" style="zoom: 67%;" />
+
+#### Step 1.1.2 - Preparing the ISO
```
Just FYI: you can also send a link to an ISO file to the technicians, who will then download it to a USB stick and connect it to your server.

n-se commented on tutorials/proxmox-docker-zfs/01.en.md:
```diff
@@ -0,0 +1,405 @@
+
+#### Step 1.1.3 - Mounting the ISO
+
+After the console is ordered, you will receive an email with instructions on how to access the console. There will be a link to the console control panel, which will have to be opened in Firefox. Chrome and Opera will not work, but your mileage may vary with other browsers. 
```
```suggestion
After the console is ordered, you will receive an email with instructions on how to access the console. There will be a link to the console control panel, which will have to be opened in your browser.
```

At least for me it also worked in Chromium. I expect it also depends on the KVM, there are Java and HTML5 KVMs.

n-se commented on tutorials/proxmox-docker-zfs/01.en.md:
```diff
@@ -0,0 +1,405 @@
+
+Sometimes, when you install using the [Rescue System](https://wiki.hetzner.de/index.php/Hetzner_Rescue-System/en), the network adapter name inside the installation VM and the host is different. This means that when you reboot, the network does not configure properly, since the adapter is named something else. Luckily, Debian uses a predictable network interface name scheme, so we can fix this qui ...
+
+Firstly, you will need to reboot back into the [Rescue System](https://wiki.hetzner.de/index.php/Hetzner_Rescue-System/en). Then, run `ip a` to see the network adapters you have installed. The one you will want to use will most likely be called `eth0` but in the case you have multiple network adapters, pick the adapter name you want to use for the Proxmox management network. Then, run the followi ...
```
You could probably also mount your partition and edit the interface configuration through the normal rescue system? Why did you choose vKVM here?

Kenzim replied:
mounting the zfs pool in rescue is tricky, the mountpoint is meant to be /, but thats already being used, so it wont mount. after you mount somewhere else, on rebooting, sometimes it wont auto remount back to / as it failed last time. i suppose you could spin up the virtual machine again and boot into the system, but thats what vkvm is there for anyway

n-se commented on tutorials/proxmox-docker-zfs/01.en.md:
````diff
@@ -0,0 +1,405 @@
+For each IP Subnet, you will need to create a new bridge. For additional IPs, you will only need one bridge for them all. For this example, I will use the additional subnet `10.0.2.0/29`. This represents 8 IP addresses, 6 of which are usable, but one will be used by the bridge, leaving only 5 for VMs. To create the bridge, add the following configuration to `/etc/network/interfaces`. You can use  ...
+
+```bash
+auto vmbr0
````
You already used vmbr0 for your main interface above

n-se commented on tutorials/proxmox-docker-zfs/01.en.md:
```diff
@@ -0,0 +1,405 @@
+
+| Field   | Value                    | Description                             |
+| ------- | ------------------------ | --------------------------------------- |
+| IP      | ``10.0.0.27``            | Where `10.0.0.27` is a an additional IP |
```
Double back ticks here

n-se commented on tutorials/proxmox-docker-zfs/01.en.md:
```diff
@@ -0,0 +1,405 @@
+
+If you don't have access to a storage box, you can skip this step.
+
+Since you now have your complementary storage box setup, you may as well use it for your ISOs. To do this, log into the web interface at `https://<10.0.0.1>:8006`, where `<10.0.0.1>` is your main IP, and select Datacentre on the left hand pane (green box). Then navigate to storage, and press Add CIFS (blue box).
```
```suggestion
Since you now have your complementary storage box setup, you may as well use it for your ISOs. To do this, log into the web interface at `https://<10.0.0.1>:8006`, where `<10.0.0.1>` is your main IP, and select Datacenter on the left hand pane (green box). Then navigate to storage, and press Add CIFS (blue box).
```

n-se commented on tutorials/proxmox-docker-zfs/01.en.md:
````diff
@@ -0,0 +1,405 @@
+For Docker to work properly on this ZFS server, we need to enable the ZFS storage driver for Docker. First, run the following commands:
+
+```bash
+service docker stop
````
can you use systemctl here too?

n-se commented on tutorials/proxmox-docker-zfs/01.en.md:
````diff
@@ -0,0 +1,405 @@
+
+```bash
+systemctl daemon-reload
+service docker start
````
can you use systemctl here too?

n-se commented on tutorials/proxmox-docker-zfs/01.en.md:
```diff
@@ -0,0 +1,405 @@
+tags: ["Development", "Proxmox", "Containers", "Docker", "Virtualization"]
+author: "Kenzi Marcel"
+author_link: "https://github.com/kenzim"
+author_img: "https://s.gravatar.com/avatar/aad2e58145359610f0bc11eec5412567?s=80&r=g"
```
@leojakeson please take a look at this, I am not sure what kind of image sources we want to accept here

Kenzim replied:
is there a preferred image source?

n-se replied:
Everyone else used the GitHub avatar, we do not yet have a guideline for profile image sources

Kenzim commented:
@n-se after reading the tutorial again, I removed the KVM part as I think it was just making the tutorial too long. Focusing on just the rescue method shortens it down a lot and simplifies a lot of things, like the network config after proxmox is installed. I think if you have access to KVM, it shouldnt be too hard for people to mount an iso and follow the normal install part of the tutorial. Also cleaned up the other things you mentioned, all in all I think it looks a lot better now

Kenzim commented:
what do you think @n-se ?

Kenzim commented:
how long until merge @n-se?
